apps/web-giddh/src/app/gst/filing/tabs/overview/view-transactions/view-transactions.component.html

```diff
@@ -1,290 +1,253 @@
-                                d="M367.5 433.7L189.8 256 367.5 78.3l22.6-22.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-223 223c-6 6-9.4 14.1-9.4 22.6 0 8.5 3.4 16.6 9.4 22.6l223 223c6.2 6.2 14.4 9.4 22.6 9.4s16.4-3.1 22.6-9.4c12.5-12.5 12.5-32.8 0-45.3l-22.5-22.6z"
-                            ></path>
-                        </svg>
-                        {{ commonLocaleData?.app_back }}
-                    </a>
-                </div>
-                <div class="filter-row" *ngIf="selectedGst === GstReport.Gstr1 && (filterParam.type === 'nil' || filterParam.type === 'b2cs')">
-                    <div class="name">{{ localeData?.filing?.transaction_type }}:</div>
-                    <div class="btn-group filter d-inline-block" dropdown>
-                        <button
-                            id="button-basic"
-                            dropdownToggle
-                            type="button"
-                            class="btn btn-primary dropdown-toggle"
-                            aria-controls="dropdown-basic"
-                        >
-                            {{ selectedFilter?.entityType }} <span class="caret"></span>
+        <button
+            mat-stroked-button
+            color="primary"
+            (click)="goBack()"
+        >
+            <svg
+                style="top: 3px; position: relative"
+                version="1.1"
+                id="Layer_1"
+                xmlns="http://www.w3.org/2000/svg"
+                x="0"
+                y="0"
+                viewBox="0 0 512 512"
+                xml:space="preserve"
+                class="icon icon-sm text-top mr-1"
+            >
+                <path
+                    d="M367.5 433.7L189.8 256 367.5 78.3l22.6-22.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-223 223c-6 6-9.4 14.1-9.4 22.6 0 8.5 3.4 16.6 9.4 22.6l223 223c6.2 6.2 14.4 9.4 22.6 9.4s16.4-3.1 22.6-9.4c12.5-12.5 12.5-32.8 0-45.3l-22.5-22.6z"
+                ></path>
+            </svg>
+            {{ commonLocaleData?.app_back }}
+        </button>
+        <div class="filter-row mr-l1" *ngIf="selectedGst === GstReport.Gstr1 && (filterParam.type === 'nil' || filterParam.type === 'b2cs')">
+            <div class="name">{{ localeData?.filing?.transaction_type }}:</div>
+            <div class="btn-group filter d-inline-block">
+                <button
+                    mat-stroked-button
+                    color="primary"
+                    type="button"
+                    [matMenuTriggerFor]="entityTypeMenu"
+                >
+                    {{ selectedFilter?.entityType }} <span class="caret"></span>
+                </button>
+                <mat-menu #entityTypeMenu="matMenu">
+                    <ng-container *ngFor="let entity of gstr1entityType">
+                        <button mat-menu-item aria-label="download" (click)="viewFilteredTxn('entityType', entity?.value)">
+                            {{ entity.label }}
```

Comment:
```suggestion
                            {{ entity?.label }}
```

Reply:
no  need it will not throw error at any condition